src/index.ts

```diff
@@ -214,6 +220,17 @@ abstract class Hook<
     };
 
     this.taps.push(tap);
+    this.taps.sort((A, B) => {
```

Comment:
Have to sort it here because the new tap being added could be in the `before` of a previous tap. So all taps need to be reevaluated.